python/ml_wrappers/model/openai_wrapper.py

```diff
@@ -0,0 +1,137 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+"""Defines a model wrapper for an openai model endpoint."""
+
+import numpy as np
+
+try:
+    import openai
+    openai_installed = True
+except ImportError:
+    openai_installed = False
+try:
+    from raiutils.common.retries import retry_function
+    rai_utils_installed = True
+except ImportError:
+    rai_utils_installed = False
+
+
+def replace_backtick_chars(message):
+    """Replace backtick characters in a message.
+
+    :param message: The message.
+    :type message: str
+    :return: The message with backtick characters replaced.
+    :rtype: str
+    """
+    return message.replace('`', '')
+
+
+class ChatCompletion(object):
+    """A class to call the openai chat completion endpoint."""
+
+    def __init__(self, messages):
+        """Initialize the class.
+
+        :param messages: The messages.
+        :type messages: list
+        """
+        self.messages = messages
+
+    def fetch(self):
+        """Call the openai chat completion endpoint.
+
+        :return: The response.
+        :rtype: dict
+        """
+        return openai.ChatCompletion.create(
+            engine="gpt-4-32k",
```

Comment:
What about other types?

Reply:
good point, done, made this a parameter on the endpoint!